lib/msf/core/payload/adapter/fetch.rb

```diff
@@ -94,15 +94,16 @@ def generate_fetch_commands
     #
     case datastore['FETCH_COMMAND'].upcase
     when 'FTP'
-      return _generate_ftp_command
+      return datastore['FETCH_FILELESS'] && linux? ? _generate_ftp_command_fileless : _generate_ftp_command
     when 'TNFTP'
-      return _generate_tnftp_command
+      return datastore['FETCH_FILELESS'] && linux? ? _generate_tnftp_command_fileless : _generate_tnftp_command
     when 'WGET'
-      return _generate_wget_command
+      return datastore['FETCH_FILELESS'] && linux? ? _generate_wget_command_fileless : _generate_wget_command
     when 'CURL'
-      return _generate_curl_command
+      return datastore['FETCH_FILELESS'] && linux? ? _generate_curl_command_fileless : _generate_curl_command
     when 'TFTP'
-      return _generate_tftp_command
+      return datastore['FETCH_FILELESS'] && linux? ? _generate_tftp_command_fileless : _generate_tftp_command
+    # ignoring certutil when FETCH_FILELESS is enabled
```

Comment:
The comment is wrong. I think something like "FETCH_FILELESS isn't supported for certutil for now" would be better.

Reply:
certutil should not be an option for linux payloads?
The `FETCH_COMMAND option comes from either lib/msf/core/payload/adapter/fetch/windows_options lib/msf/core/payload/adapter/fetch/linux_options.  Nix-y payloads should not import the windows_options file which allows the use of certutil.

Reply:
Yeah; by adding the `FETCH_FILELESS` option to `linux_options` the user cannot select it and cURL.  We already support fileless fetch execution on windows via SMB, though we don't advertise it.